src/lib/logger.ts

```diff
@@ -0,0 +1,55 @@
+// eslint-disable no-console
+import { Logger, LoggerFunction, LogLevel } from '@shapeshiftoss/logger'
+
+import { getConfig } from '../config'
+
+const logLevelCSS: Record<Exclude<LogLevel, LogLevel.NONE>, string> = {
+  [LogLevel.TRACE]: 'background-color: blue; color: white;',
+  [LogLevel.DEBUG]: 'background-color: cyan; color: black;',
+  [LogLevel.INFO]: 'background-color: green; color: white;',
+  [LogLevel.WARN]: 'background-color: yellow; color: black;',
+  [LogLevel.ERROR]: 'background-color: red; color: white;',
+}
+
+const logMessageCSS: Record<Exclude<LogLevel, LogLevel.NONE>, string> = {
+  [LogLevel.TRACE]: 'background-color: lightblue; color: darkblue;',
+  [LogLevel.DEBUG]: 'background-color: lightcyan; color: darkcyan;',
+  [LogLevel.INFO]: 'background-color: lightgreen; color: darkgreen;',
+  [LogLevel.WARN]: 'background-color: lightyellow; color: darkyellow',
+  [LogLevel.ERROR]: 'background-color: lightred; color: red;',
+}
+
+const __console = { ...console }
+
+const browserLoggerFn: LoggerFunction = (level, data) => {
+  const consoleFn = level === LogLevel.TRACE ? LogLevel.DEBUG : level
+  let msg = data._messages?.join(' ') || data.message
+
+  if (data.error) {
+    msg = `Error [Kind: ${data.error.kind}]${
+      data.error.code ? ` [Code: ${data.error?.code}]` : ''
+    } ${data.error.message} `
+  }
```

Comment:
I suspect you'll disagree, but I'd prefer to remove the mutation with either an IIFE or ternary:

```ts
const msg = (() => {
  if (data.error)
    return `Error [Kind: ${data.error.kind}]${
      data.error.code ? ` [Code: ${data.error?.code}]` : ''
    } ${data.error.message} `
  else return data._messages?.join(' ') || data.message
})()
```

or

```ts
const msg = data.error
  ? `Error [Kind: ${data.error.kind}]${data.error.code ? ` [Code: ${data.error?.code}]` : ''} ${
      data.error.message
    } `
  : data._messages?.join(' ') || data.message
```

Reply:
I like the IIFE better than the ternary FWIW.  

Anytime I see nested ternaries my brain twitches